https://github.com/PaperMC/Paper/pull/9129 Causes unkown/missing dependency errors

### Expected behavior

Expected Paper to identify and load dependencies correctly.

### Observed/Actual behavior

When attempting to start a server with EssentialsX along with EssentialsX modules, Paper is unable to identify EssentialsX as a satisfied dependency. 
Issue persists on dev and stable branch of EssentialsX.

`23.04 00:26:35 [Server] [ERROR] [LegacyPluginLoadingStrategy] Could not load 'plugins/EssentialsXSpawn-2.20.0-dev+56-fd49b70.jar' in folder 'plugins'
23.04 00:26:35 [Server] [INFO] org.bukkit.plugin.UnknownDependencyException: Unknown/missing dependency plugins: [Essentials]. Please download and install these plugins to run 'EssentialsSpawn v2.20.0-dev+56-fd49b70'.
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:115) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:35) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.strategy.LegacyPluginLoadingStrategy.loadProviders(LegacyPluginLoadingStrategy.java:200) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:39) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at org.bukkit.craftbukkit.v1_19_R3.CraftServer.loadPlugins(CraftServer.java:423) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:273) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1104) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at java.lang.Thread.run(Thread.java:831) ~[?:?]
23.04 00:26:35 [Server] [ERROR] [LegacyPluginLoadingStrategy] Could not load 'plugins/EssentialsXChat-2.20.0-dev+56-fd49b70.jar' in folder 'plugins'
23.04 00:26:35 [Server] [INFO] org.bukkit.plugin.UnknownDependencyException: Unknown/missing dependency plugins: [Essentials]. Please download and install these plugins to run 'EssentialsChat v2.20.0-dev+56-fd49b70'.
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:115) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:35) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.strategy.LegacyPluginLoadingStrategy.loadProviders(LegacyPluginLoadingStrategy.java:200) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:39) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at org.bukkit.craftbukkit.v1_19_R3.CraftServer.loadPlugins(CraftServer.java:423) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:273) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1104) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at java.lang.Thread.run(Thread.java:831) ~[?:?]
23.04 00:26:35 [Server] [ERROR] [LegacyPluginLoadingStrategy] Could not load 'plugins/EssentialsXAntiBuild-2.20.0-dev+56-fd49b70.jar' in folder 'plugins'
23.04 00:26:35 [Server] [INFO] org.bukkit.plugin.UnknownDependencyException: Unknown/missing dependency plugins: [Essentials]. Please download and install these plugins to run 'EssentialsAntiBuild v2.20.0-dev+56-fd49b70'.
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:115) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:35) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.strategy.LegacyPluginLoadingStrategy.loadProviders(LegacyPluginLoadingStrategy.java:200) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:39) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at org.bukkit.craftbukkit.v1_19_R3.CraftServer.loadPlugins(CraftServer.java:423) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:273) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1104) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]
23.04 00:26:35 [Server] [INFO] at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-"8e3009e"]`



### Steps/models to reproduce

Update to Paper-517
Install EssentialsX
Install any additional EssentialsX module
Observe issue

### Plugin and Datapack List

ClearLag, Essentials, EssentialsAntiBuild, EssentialsChat, EssentialsSpawn, floodgate, Geyser-Spigot, GriefPrevention, LuckPerms, Shopkeepers, Vault, WorldEdit

Issue persists on a reduced server with just Essentials, EssentialsAntiBuild, EssentialsChat, EssentialsSpawn installed.

### Paper version

Issue observed on git-Paper-517

Issue resolved by reverting to git-Paper-516

### Other

_No response_

--- a/patches/server/0013-Paper-Plugins.patch
+++ b/patches/server/0013-Paper-Plugins.patch
@@ -1762,10 +1762,10 @@ index 0000000000000000000000000000000000000000..a2fa8406bc3f0dcab6805633ae984d03
 +}
 diff --git a/src/main/java/io/papermc/paper/plugin/entrypoint/dependency/MetaDependencyTree.java b/src/main/java/io/papermc/paper/plugin/entrypoint/dependency/MetaDependencyTree.java
 new file mode 100644
-index 0000000000000000000000000000000000000000..1523cbb1801b937c3d40080df7d52e6cff8fc49c
+index 0000000000000000000000000000000000000000..e72bec3b0cbc41580f1b4beecae316d1c083d3e3
 --- /dev/null
 +++ b/src/main/java/io/papermc/paper/plugin/entrypoint/dependency/MetaDependencyTree.java
-@@ -0,0 +1,112 @@
+@@ -0,0 +1,117 @@
 +package io.papermc.paper.plugin.entrypoint.dependency;
 +
 +import com.google.common.graph.GraphBuilder;
@@ -1865,6 +1865,11 @@ index 0000000000000000000000000000000000000000..1523cbb1801b937c3d40080df7d52e6c
 +    @Override
 +    public boolean hasDependency(@NotNull String pluginIdentifier) {
 +        return this.dependencies.contains(pluginIdentifier);
++    }
++
++    // Used by the legacy loader -- this isn't recommended
++    public void addDirectDependency(String dependency) {
++        this.dependencies.add(dependency);
 +    }
 +
 +    @Override
@@ -2240,10 +2245,10 @@ index 0000000000000000000000000000000000000000..a9bca905eba67972e4d1b07b1d243272
 +}
 diff --git a/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java b/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java
 new file mode 100644
-index 0000000000000000000000000000000000000000..ebd193e7d9736021756223d565c8a16a7b1770d2
+index 0000000000000000000000000000000000000000..03a80151a28cdb712cf89d909188d9bc806ca3bf
 --- /dev/null
 +++ b/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java
-@@ -0,0 +1,269 @@
+@@ -0,0 +1,271 @@
 +package io.papermc.paper.plugin.entrypoint.strategy;
 +
 +import com.google.common.graph.GraphBuilder;
@@ -2292,6 +2297,7 @@ index 0000000000000000000000000000000000000000..ebd193e7d9736021756223d565c8a16a
 +            PluginMeta configuration = provider.getMeta();
 +
 +            PluginProvider<T> replacedProvider = providersToLoad.put(configuration.getName(), provider);
++            dependencyTree.addDirectDependency(configuration.getName()); // add to dependency tree
 +            if (replacedProvider != null) {
 +                LOGGER.severe(String.format(
 +                    "Ambiguous plugin name `%s' for files `%s' and `%s' in `%s'",
@@ -2323,6 +2329,7 @@ index 0000000000000000000000000000000000000000..ebd193e7d9736021756223d565c8a16a
 +                        provider.getParentSource()
 +                    ));
 +                } else {
++                    dependencyTree.addDirectDependency(provided); // add to dependency tree
 +                    String replacedPlugin = pluginsProvided.put(provided, configuration.getName());
 +                    if (replacedPlugin != null) {
 +                        LOGGER.warning(String.format(

--- a/patches/server/0013-Paper-Plugins.patch
+++ b/patches/server/0013-Paper-Plugins.patch
@@ -2245,7 +2245,7 @@ index 0000000000000000000000000000000000000000..a9bca905eba67972e4d1b07b1d243272
 +}
 diff --git a/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java b/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java
 new file mode 100644
-index 0000000000000000000000000000000000000000..03a80151a28cdb712cf89d909188d9bc806ca3bf
+index 0000000000000000000000000000000000000000..f59f48654eaa299bcac862991b1e2e622264639b
 --- /dev/null
 +++ b/src/main/java/io/papermc/paper/plugin/entrypoint/strategy/LegacyPluginLoadingStrategy.java
 @@ -0,0 +1,271 @@
@@ -2329,8 +2329,8 @@ index 0000000000000000000000000000000000000000..03a80151a28cdb712cf89d909188d9bc
 +                        provider.getParentSource()
 +                    ));
 +                } else {
-+                    dependencyTree.addDirectDependency(provided); // add to dependency tree
 +                    String replacedPlugin = pluginsProvided.put(provided, configuration.getName());
++                    dependencyTree.addDirectDependency(provided); // add to dependency tree
 +                    if (replacedPlugin != null) {
 +                        LOGGER.warning(String.format(
 +                            "`%s' is provided by both `%s' and `%s'",

